src/app/view-resource/view-resource.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, OnInit } from '@angular/core';
+import {Resource} from '../interfaces/resource';
+import {ResourceService} from '../../services/resource-service/resource.service';
+import {BsModalService} from 'ngx-bootstrap/modal';
+import {EditResourceComponent} from '../edit-resource/edit-resource.component';
+import {ProjectService} from '../../services/project-service/project.service';
+import {ActivatedRoute, Params} from '@angular/router';
+import {ToastrService} from 'ngx-toastr';
+import {of} from 'rxjs';
+
+@Component({
+  selector: 'app-view-resource',
+  templateUrl: './view-resource.component.html',
+  styleUrls: ['./view-resource.component.css']
+})
+export class ViewResourceComponent implements OnInit {
+  public resources: Resource[];
+  public projectIndex: number;
+  public projectName; string;
+
+  constructor(private activatedRoute: ActivatedRoute, private projectService: ProjectService,
+              private resourceService: ResourceService, public modalService: BsModalService, private toastr: ToastrService) { }
+
+  ngOnInit(): void {
+    let innerSubscription = of().subscribe();
```

Comment:
a. you can just use new Subscription()
b. instead of this wierdness, merge the observables of queryparams and getProject to one observable. That way you can unsibscribe to them both, and also add error handling to both etc...